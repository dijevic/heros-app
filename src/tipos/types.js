export const types = {
    login: '[auth] login',
    logOut: `[auth] logOut`
}