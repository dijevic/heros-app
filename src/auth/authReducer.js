import { types } from "../tipos/types";

export const authReducer = (state = {}, action) => {

    switch (action.type) {
        case types.login:
            return {
                ...action.payload,
                logged: true
            }
        case types.logOut:
            return { logged: false }

        default:
            return state
    }
}