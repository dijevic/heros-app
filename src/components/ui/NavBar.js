import React, { useContext } from 'react'
import { Link, NavLink, useHistory } from 'react-router-dom'
import { authContext } from '../../auth/AuthContext'
import { types } from '../../tipos/types'

export const Navbar = () => {
    const { usuario, dispatch } = useContext(authContext)
    const history = useHistory()
    const handleLogOut = () => {
        dispatch({
            type: types.logOut
        })
        history.replace('/login')

    }

    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark ">

            <Link
                className="navbar-brand"
                to="/"
            >
                Heroes
            </Link>

            <div className="navbar-collapse">
                <div className="navbar-nav">

                    <NavLink
                        activeClassName="active"
                        className="nav-item nav-link"
                        exact
                        to="/marvel"
                    >
                        Marvel
                    </NavLink>

                    <NavLink
                        activeClassName="active"
                        className="nav-item nav-link"
                        exact
                        to="/dc"
                    >
                        DC
                    </NavLink>

                    <NavLink
                        activeClassName="active"
                        className="nav-item nav-link"
                        exact
                        to="/search"
                    >
                        search
                    </NavLink>
                </div>
            </div>

            <div className="navbar-collapse collapse w-100 order-3 dual-collapse2 ">
                <ul className="navbar-nav ml-auto d-flex ">
                    <p className="nav-item nav-link">{usuario.logged && <b>{usuario.nombre}</b>}</p>
                    <button
                        className="nav-item nav-link btn btn-danger"
                        onClick={handleLogOut}
                    >
                        Logout
                    </button>
                </ul>
            </div>
        </nav>
    )
}