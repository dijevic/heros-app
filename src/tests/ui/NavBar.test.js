
import { mount } from "enzyme"
import { MemoryRouter, Router } from "react-router"
import { authContext } from "../../auth/AuthContext"
import { Navbar } from "../../components/ui/NavBar"
import { types } from '../../tipos/types'

describe('pruebas en mi componente del navBar', () => {
    const historyMock = {
        push: jest.fn(),
        location: {},
        listen: jest.fn(),
        replace: jest.fn(),
        createHref: jest.fn()
    }

    const context = {
        usuario: {
            logged: true,
            nombre: 'diego vielma'
        },
        dispatch: jest.fn()
    }

    afterEach(() => {
        jest.clearAllMocks()
    })
    const wrapper = mount(
        <authContext.Provider value={context}>
            <MemoryRouter>
                <Router history={historyMock}>
                    <Navbar />
                </Router>
            </MemoryRouter>
        </authContext.Provider>)



    test('debe hcer match con el snapshot', () => {

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('.name').text().trim()).toBe(context.usuario.nombre)

    })

    test('debe llamar a la funcion de logOut y la historyMock para reubicar la pagina', () => {

        wrapper.find('button').prop('onClick')()

        expect(context.dispatch).toHaveBeenCalledWith({ type: types.logOut })
        expect(historyMock.replace).toHaveBeenCalled()
    })
})