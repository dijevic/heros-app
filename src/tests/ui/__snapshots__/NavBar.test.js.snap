// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en mi componente del navBar debe hcer match con el snapshot 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark "
>
  <a
    className="navbar-brand"
    onClick={[Function]}
  >
    Heroes
  </a>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        search
      </a>
    </div>
  </div>
  <div
    className="navbar-collapse collapse w-100 order-3 dual-collapse2 "
  >
    <ul
      className="navbar-nav ml-auto d-flex "
    >
      <p
        className="nav-item nav-link name"
      >
        <b>
          diego vielma
        </b>
      </p>
      <button
        className="nav-item nav-link btn"
        onClick={[Function]}
      >
        Logout
      </button>
    </ul>
  </div>
</nav>
`;
